src/3box.js

```diff
@@ -461,56 +462,101 @@ class Box {
   }
 
   /**
-   * Creates a proof that links an external account to the 3Box account of the user.
+   * Creates a proof that links an ethereum address to the 3Box account of the user. If given proof, it will simply be added to the root store.
    *
-   * @param     {String}        type        The type of link (default 'ethereum')
+   * @param     {Object}    [link]                         Optional link object with type or proof
+   * @param     {String}    [link.type='ethereum-eoa']     The type of link (default 'ethereum')
+   * @param     {Object}    [link.proof ]                  URL of graphQL 3Box profile service
    */
-  async linkAddress (type = ACCOUNT_TYPES.ethereum) {
-    if (type === ACCOUNT_TYPES.ethereum) {
+  async linkAddress (link={}) {
+    if (link.proof) {
+      await this._writeAddressLink(proof)
+      return
+    }
+    if (!link.type || link.type === ACCOUNT_TYPES.ethereumEOA) {
       await this._linkProfile()
     }
   }
 
-  async linkAccount (type = ACCOUNT_TYPES.ethereum) {
+  async linkAccount (type = ACCOUNT_TYPES.ethereumEOA) {
     console.warn('linkAccount: deprecated, please use linkAddress going forward')
     await this.linkAddress(type)
   }
 
   /**
-   * Checks if there is a proof that links an external account to the 3Box account of the user.
+   * Remove given address link, returns true if successful
    *
-   * @param     {String}        type        The type of link (default ethereum)
+   * @param     {String}   address      address that is linked
    */
-  async isAddressLinked (type = ACCOUNT_TYPES.ethereum) {
-    if (type === ACCOUNT_TYPES.ethereum) {
-      return Boolean(await this.public.get('ethereum_proof'))
+  async removeAddressLink (address) {
```

Comment:
We also need to remove the `'link-address'` entry from the rootStore in this method.